### PR TITLE
Clean up CheckboxField

### PR DESCRIPTION
**Pull request by assertchris**

### CODE REVIEW

**Comment by assertchris** [comment]
@tractorcow ping

File: forms/CheckboxField.php [hunk 1]
```diff
@@ -6,59 +7,102 @@
...
+	 */
 	public function getAttributes() {
-		$attrs = parent::getAttributes();
-		$attrs['value'] = 1;
+		$attributes = array(
+			'type' => 'checkbox',
+			'value' => 1,
+		);
+
+		if($value = $this->Value()) {
```
Comment (tractorcow) [inline forms/CheckboxField.php]: `$value` is unused, so let's remove it?

**Comment by assertchris** [comment]
Fixed

File: forms/CheckboxField.php [hunk 2]
```diff
@@ -6,59 +7,102 @@
...
 	public function getAttributes() {
-		$attrs = parent::getAttributes();
-		$attrs['value'] = 1;
+		$attributes = array(
+			'type' => 'checkbox',
+			'value' => 1,
+		);
+
+		if($this->Value()) {
+			$attributes['checked'] = 'checked';
```
Comment (tractorcow) [inline forms/CheckboxField.php]: I just noticed; Prior to this change 'checked' would be null if not selected, but now it's not even assigned. Should we assign null in an else?

**Comment by assertchris** [comment]
Like that?

**Comment by assertchris** [comment]
@tractorcow ping

**Comment by tractorcow** [comment]
I think we should start putting these minor annoyance fixes into 3.2 branch; We aren't going to be merging up from 3.1 to 3.2 if we can help it.

**Pull request closed without merging**